### ISSUE TITLE
False positive for "Null pointer exception may occur here" inspection

### ISSUE THREAD

**Issue by FlyingDR** [kalessil/phpinspectionsea#1118]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Positive       |
| Plugin         | Php Inspections (EA Extended), v3.0.8.1 |
| Language level | PHP 7.1, 7.2, 7.3                      |

Current behaviour
--
Consider following simplified code example:

```php
class Item
{
    public function test(): bool
    {
        return true;
    }
}

class Collection
{
    /**
     * @var Item[]
     */
    private $items;

    public function find(callable $test): ?Item
    {
        // Actually contents of this method doesn't matters, added as example of possible method contents
        return array_reduce($this->items, function ($found, $current) use ($test) {
            if ($found !== null) {
                return $found;
            }
            return $test($current) ? $current : null;
        });
    }
}

class App
{
    public function test(): void
    {
        $item = (new Collection())->find(function (Item $item) {
            // "Null pointer exception may occur here" warning for $item variable
            return $item->test();
        });
    }
}

function test() {
    $item = (new Collection())->find(function (Item $item) {
        // "Null pointer exception may occur here" warning for $item variable
        return $item->test();
    });
}

$item = (new Collection())->find(function (Item $item) {
    // No warning
    return $item->test();
});
```
We have a very simple `Item` class and very simple `Collection` with ability to search for `Item`. In some cases PHPStorm reports "Null pointer exception may occur here" for `$item` variable inside callable function. 

Report is triggered in a case if same variable name is used both inside callable and outside it, but only in a case if callable itself is located inside class method or standalone function.

Expected behaviour
--
This report is false positive because outer variable and variable inside callable are different variables and can have same name.

Environment details
---
PhpStorm 2018.3
Build #PS-183.4284.150, built on November 21, 2018
JRE: 1.8.0_152-release-1343-b15 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 7 6.1

**Comment by FlyingDR** [comment]
I've published example with other test variants as a [gist](https://gist.github.com/FlyingDR/f2fe8f79d16d01aa6ce6f5e7a39676a7) for reference

**Comment by kalessil** [comment]
Thank you for detailed report @FlyingDR, I'll check the issue.

**Comment by kalessil** [comment]
Thank you for reporting @FlyingDR, I'll take care of fixing the issue.

**Comment by kalessil** [comment]
Fixed!

**Comment by pthreat** [comment]
2020, have the exact same issue, null pointer exceptions in PHP

**Comment by kalessil** [comment]
@pthreat: thank you for reporting. Please file a new issue with an example, this issue has been closed more tham year ago. Thank you in advance.

**Comment by Chemaclass** [comment]
@kalessil, I created a follow-up issue for a possible false-positive: https://github.com/kalessil/phpinspectionsea/issues/1814
